app/code/core/Mage/Catalog/Model/Product/Image.php

```diff
@@ -183,7 +183,7 @@ protected function _getMemoryLimit()
     {
         $memoryLimit = trim(strtoupper(ini_get('memory_limit')));
 
-        if (!isSet($memoryLimit[0])){
+        if (!isSet($memoryLimit)){
```

Comment:
I don't think this is doing what you think it's doing. 
`$memoryLimit = trim(strtoupper(ini_get('memory_limit')));` will always set the variable. As far as I can tell, `!empty($memoryLimit)` is the proper check in this case.